editor/editor_properties.cpp

```diff
@@ -2060,7 +2052,12 @@ EditorPropertyBasis::EditorPropertyBasis() {
 	g->set_columns(3);
 	add_child(g);
 
-	static const char *desc[9] = { "x", "y", "z", "x", "y", "z", "x", "y", "z" };
+	static const char **desc;
+	if (EDITOR_GET("interface/inspector/use_vector_component_notation_for_matrices")) {
+		desc = new const char *[9] { "x.x", "y.x", "z.x", "x.y", "y.y", "z.y", "x.z", "y.z", "z.z" };
```

Comment:
Shouldn't we use `Array()`/`PackedStringArray()` instead of C-style arrays created with `new` and `delete`?

Alternatively, you can remove the editor setting since it's disabled by default. I wouldn't add an editor setting for something that doesn't have significant demand (i.e. don't future-proof settings).

Reply:
I don't know how to create a PackedStringArray with values already in it.

Anyway, I removed the setting, you're right that it's not necessary.